src/controller/src/lib.rs

```diff
@@ -540,7 +544,8 @@ where
         + From<EpochMillis>
         + TimestampManipulation
         + std::fmt::Display
-        + Into<Datum<'static>>,
+        + Into<Datum<'static>>
+        + Into<mz_repr::Timestamp>,
```

Comment:
It's needed, so I'll remove the other bound. There might be other ways of expressing the requirements, but I'll leave as is, because it works for now. 😅